### PR TITLE
tools: add ClinicalTrials.gov API as function tool

### PR DESCRIPTION
**Pull request by AidanCooper**
Thank you for contributing to the Cohere Toolkit!

- [x] **PR title**: "area: description"

Enables clinical studies to be retrieved from ClinicalTrials.gov, with optional filtering by condition, location, intervention, and recruiting status.
Screenshot

- [x] **Add tests and docs**: Please include testing and documentation for your changes
- [x] **Lint and test**: Run `make lint` and `make run-tests` 

Three unrelated tests are failing locally for me:
```
FAILED src/backend/tests/routers/test_deployment.py::test_list_deployments_has_all_option[mock_available_model_deployments0] - AssertionError: assert 5 == 3
FAILED src/backend/tests/routers/test_deployment.py::test_list_deployments_no_available_models_404[mock_available_model_deployments0] - assert 200 == 404
FAILED src/backend/tests/routers/test_deployment.py::test_list_deployments_no_available_models_with_all_option[mock_available_model_deployments0] - AssertionError: assert 5 == 3
```

### CODE REVIEW

**Review by lusmoura — APPROVED** [review]
Great addition to the toolkit! Thank you for contributing 😄 You might need to update your branch before merging it.

**Comment by AidanCooper** [comment]
> Great addition to the toolkit! Thank you for contributing 😄 You might need to update your branch before merging it.

Thanks! Branch is up to date — just need someone with write access to merge

**Pull request merged**
